Test suite

ChiragKasat opened the pull request.

ChiragKasat committed:
update requirements.txt and create tests module

ChiragKasat committed:
Merge branch 'master' of https://git.postgresql.org/git/pgweb into test-suite
merging

ChiragKasat committed:
fix versions of packages, write a testing doc, and written additional
unit tests

ChiragKasat committed:
create conftest.py file and segregate the requiremnts.txt file to testing_requirements

ChiragKasat committed:
Merge branch 'master' of https://git.postgresql.org/git/pgweb into test-suite

ChiragKasat committed:
change functions' name from camel case to snake case

ChiragKasat committed:
use factory boy to test models and written tests for core models

ChiragKasat committed:
written more core models unit tests

ChiragKasat committed:
write tests for core and make factories of news models

andreasscherbaum commented on conftest.py:
```diff
@@ -0,0 +1,31 @@
+import pytest
+from django.db import connections
+import psycopg2
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT\
```
Is there a reason for the backslash at the end?

andreasscherbaum commented on conftest.py:
```diff
@@ -0,0 +1,31 @@
+def django_db_setup():
+    from django.conf import settings
+
+    settings.DATABASES['default']['NAME'] = 'the_copied_db'
```
Might be useful to use a more descriptive name here. "copied db" is meaningless.

andreasscherbaum commented on pytest.ini:
```diff
@@ -0,0 +1,4 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = pgweb.settings
+python_files = tests.py test_*.py *_tests.py
+addopts = -p no:warnings
```
There is a git warning.

andreasscherbaum commented on tests/core/factories.py:
```diff
@@ -0,0 +1,111 @@
+        model = ImportedRSSFeed
+
+    internalname = "planet"
+    url = "https://www.postgresql.org/"
```
Planet is at: planet.postgresql.org, not at www

andreasscherbaum commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,147 @@
+@pytest.mark.django_db
+class TestOrganisationModel(TestCase):
+    def setUp(self):
+        self.user = UserFactory(first_name="John", last_name="Doe", email="johndoe@test.com")
```
There is the .invalid TLD (defined in RFC 2606) which should be used instead of valid domain names like "test.com".

ChiragKasat replied:
is @examaple.com fine?

andreasscherbaum replied:
No, that's also a valid domain.

ChiragKasat committed:
minor changes and change name of test db

mhagander committed:
Remove final(?) mention of Linux graphical installer

e836d33bbfb1ebe5530acdd75eeb5f00a5ddbfbf tried to remove most of it, but
left a danglig sentence on the "other linux" page.

ChiragKasat committed:
fix email issue and add news app models' tests

ChiragKasat committed:
add more tests for news app models

ChiragKasat committed:
write reset password view tests

ChiragKasat committed:
Merge branch 'master' of https://git.postgresql.org/git/pgweb into test-suite

ChiragKasat committed:
add merge and pending view tests

markrofail commented on pgweb/core/models.py:
```diff
@@ -71,7 +71,7 @@ def treestring(self):
             return "%s %s" % (self.numtree, TESTING_SHORTSTRING[self.testing])
 
-    def save(self):
+    def save(self, *args, **kwargs):
```
Why did you change this?
Please stay away from the source code

markrofail commented on tests/core/factories.py:
```diff
@@ -0,0 +1,126 @@
+    class Meta:
+        model = Version
+
+    tree = 13.3
+    latestminor = 0
+    reldate = datetime.datetime.today()
+    current = False
+    supported = True
+    testing = 0
+    docsloaded = datetime.datetime.now()
+    firstreldate = datetime.datetime.today()
+    eoldate = datetime.datetime.today()
+
+
+class QuoteFactory(DjangoModelFactory):
+    class Meta:
+        model = Quote
+
+    approved = False
+    quote = "Test"
+    who = "test name"
+    org = "PostgreSQL"
+    link = "https://postgresql.org"
```
Why not use some `Faker` methods to randomize?

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+    def test_quote_display_name(self):
+        self.assertEquals(str(self.quote), self.quote.quote)
+
+    def test_max_quote_display_name_length(self):
+        self.assertNotEquals(str(self.largerQuote), self.largerQuote.quote)
+        self.assertEquals(len(str(self.largerQuote)), 78)
+        self.assertEquals(str(self.largerQuote)[-3:], '...')
```
I dont understand the purpose of this test

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+class TestVersionModel(TestCase):
+    def setUp(self):
+        self.version = VersionFactory()
+        self.testing_version_rc = VersionFactory(tree=12.1, testing=1, latestminor=1)
+        self.testing_version_beta = VersionFactory(tree=12.2, testing=2, latestminor=2)
+        self.testing_version_alpha = VersionFactory(tree=12.3, testing=3, latestminor=3)
```
No need to create these in the `setUp` do it in each test seperatley

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+        self.assertIsInstance(self.version, Version)
+
+    def test_version_greater_than_10(self):
+        assert str(self.version) == "13.0"
+        assert self.version.treestring == "13"
+        assert self.version.relnotes == "release-13-0.html"
```
please be consistent, either use `assert` or `self.assert()`

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+
+@pytest.mark.django_db
+class TestUserProfileModel(TestCase):
+    def setUp(self):
+        self.user = UserProfileFactory()
```
No need to setUp. please regulate this on all the testcases

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+
+
+@pytest.mark.django_db
+def test_country_model():
+    country = Country.objects.create(name="Test Country", tld="TST")
+    assert str(country) == country.name
```
Please be consistent, either functional tests or class based tests

markrofail replied:
Why not a `CountryFactory`

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+
+@pytest.mark.django_db
+def test_moderation_notification_model():
+    notif = ModerationNotification.objects.create(objectid=1,
```
use a factory here.

markrofail commented on tests/core/test_models.py:
```diff
@@ -0,0 +1,151 @@
+
+@pytest.mark.django_db
+def test_language_model():
+    language = Language.objects.create(alpha3="test",
```
use a factory here.

markrofail commented on tests/core/test_views.py:
```diff
@@ -0,0 +1,159 @@
+    def setUp(self):
+        return super(TestAdminMergeOrgView, self).setUp()
+
+    def test_login_required(self):
+        url = '/admin/mergeorg/'
+        response = self.user_client.get(url)
+        self.assertIsInstance(response, HttpResponseRedirect)
+        self.assertEquals(response.status_code, 302)
+        self.assertEquals(response.url, '/account/login/?next=' + url)
+
+    def test_user_redirected(self):
+        url = '/admin/mergeorg/'
+        response = self.user_client.get(url)
+        self.assertEquals(response.status_code, 302)
+        self.assertIsInstance(response, HttpResponseRedirect)
+        self.assertEquals(response.url, '/account/login/?next=' + url)
```
What's the difference between these two tests?

markrofail commented on tests/core/test_views.py:
```diff
@@ -0,0 +1,159 @@
+        self.user.set_password('user1')
+        self.user.save()
+        user_login = self.user_client.login(username='user1', password='user1')
+        self.assertTrue(user_login)
```
what does the `assertTrue` accomplish here?

markrofail commented on tests/core/test_views.py:
```diff
@@ -0,0 +1,159 @@
+        self.admin_user.save()
+        admin_login = self.admin_client.login(username='admin', password='admin')
+        self.assertTrue(admin_login)
+        self.staff_user = UserFactory.build(username='staff', is_staff=True)
```
why are you using `UserFactory.build` and `UserFactory.create` sometimes?
always use `UserFactory.create`
